src/reference-manual/types.Rmd

```diff
@@ -870,53 +879,69 @@ For constructing arrays in Stan, see [Vector, Matrix, and Array Expressions](#ve
 
 ### Declaring array variables {-}
 
-Arrays are declared by enclosing the dimensions in square brackets
-following the name of the variable.
+Arrays are declared with the keyword `array` followed by the dimensions
+enclosed in square brackets and the array type and the name of the variable.
 
 The variable `n` is declared as an array of five integers as follows.
 
 ```stan
-int n[5];
+array[5] int n;
 ```
 
 A two-dimensional array of real values with three rows and four columns is
 declared with the following.
 
 ```stan
-real a[3, 4];
+array[3, 4] real a;
 ```
 
 A three-dimensional array `z` of positive reals with five rows, four
 columns, and two shelves can be declared as follows.
 
 ```stan
-real<lower=0> z[5, 4, 2];
+array[5, 4, 2] real<lower=0> z;
 ```
 
 
 Arrays may also be declared to contain vectors.  For example,
 
 ```stan
-vector[7] mu[3];
+array[3] vector[7] mu;
 ```
 
 declares `mu` to be an array of size 3 containing vectors with 7
 elements.  Arrays may also contain matrices.  The example
 
 ```stan
-matrix[7, 2] mu[15, 12];
+array[15, 12] matrix[7, 2] mu;
 ```
 
 declares a 15 by 12 array of $7 \times 2$ matrices.  Any of the
 constrained types may also be used in arrays, as in the declaration
 
 ```stan
-cholesky_factor_cov[5, 6] mu[2, 3, 4];
+array[2, 3, 4] cholesky_factor_cov[5, 6] mu;
 ```
 
 of a $2 \times 3 \times 4$ array of $5 \times 6$ Cholesky factors of
 covariance matrices.
 
+#### Old array syntax
```

Comment:
Is it old or deprecated?

Reply:
Its not deprecated yet (it will not warn). I guess we will deprecate it once tuples are introduced, You will not be able to define an arrays of tuples with the old syntax anymore. For now there is no difference between the two.